### PR TITLE
Dockerize

### PR DESCRIPTION
**Pull request by ollielo**
@junghans 

I think it is good enough to consider to be merge into the master branch. It passes all "tests" and there are Docker images on the DockerHub.

### CODE REVIEW

**Comment by junghans** [comment]
I not 100% sure what you are trying to achieve with this dockerization? If you are using the latest fedora lapacke, scotch and metis can be provided by system packages and don't need to be build. Which leaves us with exodus (losalamos/flecsi#74) and legion.

File: dockerfiles/fedora [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+RUN groupadd -r flecsi && useradd -r -m -g flecsi flecsi
+USER flecsi
+
+WORKDIR /tmp
+RUN git clone --recursive --depth=1 https://github.com/losalamos/flecsi-third-party tpl; \
+    mkdir -p tpl/build && cd tpl/build; \
+    cmake -DENABLE_EXODUS=ON -DENABLE_LAPACK=ON \
+          -DENABLE_METIS=ON  -DENABLE_SCOTCH=ON \
+          -DENABLE_LEGION=ON -DCMAKE_INSTALL_PREFIX=$HOME/tpl .. && make -j2; \
+    cd -;
```
Comment (junghans) [inline dockerfiles/fedora]: You also want to build flecsi itself to check if you changes in tpl break flecsi - see the current travis.yml file.

File: .travis.yml [hunk 2]
```diff
@@ -1,39 +1,23 @@
...
-  global:
-    - CCACHE_CPP2=yes
-    - GVER=5
   matrix:
-    - SHARED=ON METIS_INT64=OFF
-    - SHARED=ON METIS_INT64=ON
+    - DOCKERFILE=fedora
+    - DOCKERFILE=ubuntu
+#    - SHARED=ON METIS_INT64=OFF
+#    - SHARED=ON METIS_INT64=ON
```
Comment (junghans) [inline .travis.yml]: Why is this disabled?

Comment (ollielo) [inline .travis.yml]: I though you said you don't care if SHARED is on or off.

Comment (junghans) [inline .travis.yml]: But I care about `METIS_INT64`!

File: dockerfiles/fedora (hunk 1 above)
Comment (ollielo) [inline dockerfiles/fedora]: Do we really want that kind of build process? I assume that we will push to the Flecsi repo frequently enough that will trigger the "build Flecsi after new TPL" without doing it here.

Comment (ollielo) [inline dockerfiles/fedora]: Also, what version do we want to push to DockerHub?

Comment (junghans) [inline dockerfiles/fedora]: Well, can we not just make two builds and only push one to docker hub?

Comment (ollielo) [inline dockerfiles/fedora]: Sure, but which one?

Comment (junghans) [inline dockerfiles/fedora]: I would push the image without flecsi in it.

**Comment by ollielo** [comment]
@junghans 

Flecsi TPL is pushing an image with Flecsi built-in. We have to merge this PR before the Flecsi [PR 89](https://github.com/losalamos/flecsi/pull/89).

**Pull request merged**